From 2c8f0263382bf64800faec5fba5cc3e005d9fb1e Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Thu, 31 Jan 2019 10:54:00 +0000
Subject: [PATCH] Update for change to access_ok in Linux 5.0

Linux 5.0 removed the 1st argument 'type' from the access_ok macro.
Update the ppm_access_ok() macro to cater for this change for Linux
5.0

Bug: https://github.com/draios/sysdig/issues/1299
sysdig-CLA-1.0-signed-off-by: Colin Ian King <colin.king@canonical.com>

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 driver/ppm_events.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/driver/ppm_events.c b/driver/ppm_events.c
index 717590888..cc3eb98d2 100644
--- a/driver/ppm_events.c
+++ b/driver/ppm_events.c
@@ -46,7 +46,11 @@ or GPL2.txt for full copies of the license.
 #ifdef access_ok_noprefault
 #define ppm_access_ok access_ok_noprefault
 #else
-#define ppm_access_ok access_ok
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 0, 0)
+#define ppm_access_ok(type, addr, size)	access_ok(type, addr, size)
+#else
+#define ppm_access_ok(type, addr, size)	access_ok(addr, size)
+#endif
 #endif
 
 extern bool g_tracers_enabled;
